nexus-ingest/nexus-sink/src/main/java/org/nasa/jpl/nexus/ingest/nexussink/NexusSinkOptionsMetadata.java

```diff
@@ -114,22 +122,77 @@ public void setSolrCollection(String solrCollection){
     @ModuleOption(value = "number of messages to buffer before inserting into Nexus", defaultValue = "0")
     public void setInsertBuffer(Integer insertBuffer){ this.insertBuffer = insertBuffer; }
 
+    @ModuleOption(value = "The name of the S3 bucket", defaultValue = "nexus-jpl")
+    public void setS3BucketName(String s3BucketName) {
+        this.s3BucketName = s3BucketName;
+    }
+
+    public String getS3BucketName() {
+        return this.s3BucketName;
+    }
+
+    @ModuleOption(value = "The AWS region", defaultValue = "us-west-1")
+    public void setAwsRegion(String awsRegion) {
+        this.awsRegion = awsRegion;
+    }
+
+    public String getAwsRegion() {
+        return this.awsRegion;
+    }
+
+    @ModuleOption(value = "The name of the dynamoDB table", defaultValue = "silvan9145table")
+    public void setDynamoTableName(String dynamoTableName) {
+        this.dynamoTableName = dynamoTableName;
+    }
+
+    public String getDynamoTableName() {
+        return this.dynamoTableName;
+    }
+
+    @AssertTrue(message = "Either "+PROPERTY_NAME_CASSANDRA_KEYSPACE+", "+PROPERTY_NAME_S3_BUCKET+", or "
+            +PROPERTY_NAME_DYNAMO_TABLE_NAME+" is allowed but not more than 1.")
+    public boolean isOptionMutuallyExclusiveDataStore(){
+        return Exclusives.atMostOneOf(StringUtils.isNotEmpty(getCassandraKeyspace()), StringUtils.isNotEmpty(getS3BucketName()),
+                StringUtils.isNotEmpty(getDynamoTableName()));
+    }
 
```

Comment:
You also need to validate that if `cassandraKeyspace` is set, `cassandraContactPoints` is also set. Similar for S3 and Dynamo. If `s3BucketName` is set, `awsRegion` is required. If `dynamoTableName` is set `awsRegion` is required.

_I think_ these could be 3 new methods all annotated with `@AssertTrue`. For example:

```
@AssertTrue(message = "Both "+PROPERTY_NAME_CASSANDRA_KEYSPACE+" and "+ PROPERTY_NAME_CASSANDRA_CONTACT_POINTS+" are required if using cassandra")
public boolean isCassandraConfigured(){
    return StringUtils.isNotEmpty(getCassandraContactPoints()) && StringUtils.isNotEmpty(getCassandraKeyspace());
}
```